src/examples/importing-function.md

```diff
@@ -1,29 +1,45 @@
-# Importing function from host environment
+# Importing a function from host environments
 
-You can import a function from host environment using special attribute and linker option.
+You can import a function from your host environment using the integration of Swift Package Manager
+with C targets. Firstly, you should declare a signature for your function in a C header with an
+appropriate `__import_name__` attribute:
 
-```swift
-@_silgen_name("add")
-func add(_: Int, _: Int) -> Int
+```c
+__attribute__((__import_name__("add")))
+extern int add(int lhs, int rhs);
+```
+
+Here `__import_name__` specifies the name under which this function will be exposed to Swift code.
+Move this C header to a separate target, we'll call it `HostFunction` in this example. Your
+`Package.swift` manifest for your WebAssembly app would look like this:
 
-print("2 + 3 = \(add(2, 3))")
+```swift
+// swift-tools-version:5.3
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+import PackageDescription
+
+let package = Package(
+    name: "SwiftWasmApp",
+    targets: [
+      .target(name: "HostFunction", dependencies: []),
+      .target(name: "SwiftWasmApp", dependencies: ["HostFunction"]),
+    ]
+)
 ```
 
-You need to compile the Swift code with linker option `--allow-undefined`.
+Place this header into the `include` subdirectory of your `HostFunction` target directory. You can
+then import this host function module into Swift code just as any other module:
 
+```swift
+import HostFunction
 
-```bash
-$ TOOLCHAIN_PATH=$(dirname $(swiftenv which swiftc))/../
-$ swiftc \
-    -target wasm32-unknown-wasi \
-    -sdk $TOOLCHAIN_PATH/share/wasi-sysroot \
-    lib.swift -o lib.wasm \
-    -Xlinker --allow-undefined
+print(add(2, 2))
 ```
 
 Then, you can inject a host function into the produced WebAssembly binary.
 
-Note that we use `env` as default import module name. We will add a way to specify module name to import in the near future.
+Note that we use `env` as default import module name. We will add a way to specify module name to
+import in the near future.
```

Comment:
Module name can be specified using clang attribute.

https://github.com/swiftwasm/JavaScriptKit/blob/26a83a61b2370bba7d15cf34e726a145507ffc0f/Sources/_CJavaScriptKit/include/_CJavaScriptKit.h#L78